Python test `test_new_sharing_trigger_event` sometime lose its backend connection on Windows

See the CI job that failed: https://github.com/Scille/parsec-cloud/actions/runs/4362172689/jobs/7626762231

```python
    @pytest.mark.trio
    async def test_new_sharing_trigger_event(
        alice_core: LoggedCore,
        bob_core: LoggedCore,
        running_backend,
    ):
        KEY = SecretKey.generate()
        # First, create a folder and sync it on backend
        with freeze_time("2000-01-01", devices=[alice_core.device], freeze_datetime=True):
            wid = await alice_core.user_fs.workspace_create(EntryName("foo"))
        workspace = alice_core.user_fs.get_workspace(wid)
        with freeze_time("2000-01-02", devices=[alice_core.device], freeze_datetime=True):
            await workspace.sync()
    
        # Now we can share this workspace with Bob
        with bob_core.event_bus.listen() as spy:
            with freeze_time("2000-01-03", devices=[alice_core.device], freeze_datetime=True):
                await alice_core.user_fs.workspace_share( # <- this line get the BackendOffline exception
                    wid, recipient=UserID("bob"), role=WorkspaceRole.MANAGER
                )
```

The exception code snipet

```python
         # Actually send the command to the backend
        try:
            rep = await self.backend_cmds.realm_update_roles(
                role_certificate, ciphered_recipient_message
            )
    
        except BackendNotAvailable as exc:
>           raise FSBackendOfflineError(str(exc)) from exc
E           parsec.core.fs.exceptions.FSBackendOfflineError

parsec\core\fs\userfs\userfs.py:945: FSBackendOfflineError
```

--- a/tests/core/test_messages_monitor.py
+++ b/tests/core/test_messages_monitor.py
@@ -10,7 +10,7 @@ from parsec._parsec import CoreEvent, DateTime
 from parsec.api.data import EntryName, PingMessageContent
 from parsec.api.protocol import UserID
 from parsec.core.backend_connection import BackendConnStatus
-from parsec.core.logged_core import LoggedCore
+from parsec.core.logged_core import LoggedCore, UserFS
 from parsec.core.types import WorkspaceEntry, WorkspaceRole
 from parsec.crypto import SecretKey
 from tests.common import create_shared_workspace, freeze_time
@@ -88,22 +88,22 @@ async def test_process_while_offline(
 
 @pytest.mark.trio
 async def test_new_sharing_trigger_event(
-    alice_core: LoggedCore,
+    alice_user_fs: UserFS,
     bob_core: LoggedCore,
     running_backend,
 ):
     KEY = SecretKey.generate()
     # First, create a folder and sync it on backend
-    with freeze_time("2000-01-01", devices=[alice_core.device], freeze_datetime=True):
-        wid = await alice_core.user_fs.workspace_create(EntryName("foo"))
-    workspace = alice_core.user_fs.get_workspace(wid)
-    with freeze_time("2000-01-02", devices=[alice_core.device], freeze_datetime=True):
+    with freeze_time("2000-01-01", devices=[alice_user_fs.device], freeze_datetime=True):
+        wid = await alice_user_fs.workspace_create(EntryName("foo"))
+    workspace = alice_user_fs.get_workspace(wid)
+    with freeze_time("2000-01-02", devices=[alice_user_fs.device], freeze_datetime=True):
         await workspace.sync()
 
     # Now we can share this workspace with Bob
     with bob_core.event_bus.listen() as spy:
-        with freeze_time("2000-01-03", devices=[alice_core.device], freeze_datetime=True):
-            await alice_core.user_fs.workspace_share(
+        with freeze_time("2000-01-03", devices=[alice_user_fs.device], freeze_datetime=True):
+            await alice_user_fs.workspace_share(
                 wid, recipient=UserID("bob"), role=WorkspaceRole.MANAGER
             )
 
@@ -134,7 +134,9 @@ async def test_new_sharing_trigger_event(
 
 
 @pytest.mark.trio
-async def test_revoke_sharing_trigger_event(alice_core, bob_core, running_backend):
+async def test_revoke_sharing_trigger_event(
+    alice_user_fs: UserFS, bob_core: LoggedCore, running_backend
+):
     KEY = SecretKey.generate()
 
     def _update_event(event):
@@ -148,11 +150,11 @@ async def test_revoke_sharing_trigger_event(alice_core, bob_core, running_backen
         return event
 
     with freeze_time("2000-01-02"):
-        wid = await create_shared_workspace(EntryName("w"), alice_core, bob_core)
+        wid = await create_shared_workspace(EntryName("w"), alice_user_fs, bob_core)
 
     with bob_core.event_bus.listen() as spy:
         with freeze_time("2000-01-03"):
-            await alice_core.user_fs.workspace_share(wid, recipient=UserID("bob"), role=None)
+            await alice_user_fs.workspace_share(wid, recipient=UserID("bob"), role=None)
 
         # Each workspace participant should get the message
         await spy.wait_with_timeout(

